benchmarks/nemo/tracer_advection/original/Makefile

```diff
@@ -46,14 +50,24 @@ tra_adv_omp: ../scripts/omp_trans.py
 	${MAKE} PROF_LIB_INC="-I../${DL_TIMER_DIR}/src" \
             LDFLAGS="${LDFLAGS} ../${DL_TIMER_DIR}/${DL_TIMER_NAME}" -C $@
 
+tra_adv_omp_gpu: ../scripts/omp_gpu_trans.py
+	mkdir -p $@
+	${PSYCLONE} -s ../scripts/omp_gpu_trans.py -opsy $@/tra_adv.f90 ./tra_adv.F90
+	cp Makefile_gen $@/Makefile
+	${MAKE} PROF_LIB_INC="-I../${DL_TIMER_DIR}/src" \
+            LDFLAGS="${LDFLAGS} ${OMPTARGETFLAGS} ../${DL_TIMER_DIR}/${DL_TIMER_NAME}" -C $@
```

Comment:
I would expect:
```
source <compiler_setup>/nvidia.sh
make tra_adv_omp_gpu
```
to work out of the box. But it doesn't because it requires the manage memory flag and manually update some environment variable.

We could add a new envvar in nvidia.sh
`export OMPTARGET_UM="-gpu=managed"`
(UM as unified memory because its a more standard term than managed memory across compilers)

and then add:
```
     cp Makefile_gen $@/Makefile
ifndef OMPTARGET_UM 
    $(error This implementation requires OpenMP unified memory, but the OMPTARGET_UM environment variable is undefined)
endif
     ${MAKE} PROF_LIB_INC="-I../${DL_TIMER_DIR}/src" \
```
This would allow it to work for nvidia directly but also provide and useful error message for other compilers instead of letting them compile and fail at runtime.

Reply:
That's a good idea and I've done it. I've also altered the way the various variables are passed down to the `Makefile_gen` to make it a bit clearer what's going on.